app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala

```diff
@@ -124,8 +124,13 @@ class SecureMessageService @Inject()(
   private def forwardMessage(conversationId: String, messagesRequest: CustomerMessage)(
     implicit ec: ExecutionContext,
     request: Request[_]): EitherT[Future, SecureMessageError, Unit] = {
+    val ACKNOWLEDGEMENT_REFERENCE_MAX_LENGTH = 32
```

Comment:
why you need max 32 where on line 134 govuk-tax is added ?

Reply:
I'm getting a validation error when testing in FIT1 with that reference being more than 32 characters.

Reply:
does X-Correlation-ID have the same prefix you're adding to the requestId: govuk-tax- ?

Reply:
no, this is about `acknowledgementReference`, which needs to be 32 char max, I didn't change the code for requestId, that's existing code for another story. I think maybe the auditing story Rob worked on.